### PR TITLE
BUG: Fix owens_t function when a tends to infinity

### LINKED ISSUE #15117: BUG:  owens_t(h, infinity) is not correct
### Describe your issue.

Hello,

 `owens_t(0, inf)` gives `0.5`. However `owens_t(a, inf)` is `atan(a)/(2*pi)` , so  `owens_t(0, inf)` shoud give `0.25`.  

### Reproducing Code Example

```python
from scipy import special
from math import atan, pi, inf

special.owens_t(0,inf)
# 0.5
atan(inf)/2/pi
# 0.25
```

### Error message

```shell
No error message
```

### SciPy/NumPy/Python version information

scipy 1.7.1, python 3.9.5

### PR DESCRIPTION
**Pull request by NamamiShanker**
#### Reference issue

Closes #15117 

#### What does this implement/fix?

Fixed owens_t(h, a) function giving wrong answers when a tends to infinity.

#### Additional information

Added and modified tests.

### CODE REVIEW

File: scipy/special/tests/test_owens_t.py [hunk 1]
```diff
@@ -28,8 +28,15 @@ def test_nans():
...
 def test_infs():
+    h, a = 0, np.inf
+    # T(0, a) = 1/2π * arctan(a)
+    res = 1/(2*np.pi) * np.arctan(a)
+    assert_allclose(sc.owens_t(h, a), res, rtol=5e-14)
+    assert_allclose(sc.owens_t(h, -a), -res, rtol=5e-14)
+
     h = 1
-    res = 0.5*sc.erfc(h/np.sqrt(2))
+    # Value approximated through Numerical Integration
```
Comment (tirthasheshpatel) [inline scipy/special/tests/test_owens_t.py:38]: Can you also mention what you integrated? Did you use `scipy.integrate.quad` for numerical integration?

Comment (NamamiShanker) [reply scipy/special/tests/test_owens_t.py:38]: Yes, I used scipy.integrate.quad. I computed the value of the T function integral form for the given 'h' and 'a' values. I will mention the resulting expression

Comment (NamamiShanker) [reply scipy/special/tests/test_owens_t.py:40]: I have added the integrand function and the reference. Please review it.

Comment (tirthasheshpatel) [reply scipy/special/tests/test_owens_t.py:40]: The comment is very helpful, thanks. I verified the integral from Wikipedia and it looks correct.

**Review by tirthasheshpatel — APPROVED** [review]
LGTM now. Thanks, @NamamiShanker! Tests look strong enough and C code changes look correct. @WarrenWeckesser Do you want to take a look at this since you commented on the issue?

**Review by WarrenWeckesser — APPROVED** [review]
Looks good, thanks @NamamiShanker.

**Pull request merged**